File: Python/Algorithms/RecursionAlgorithms/factorial_.py
"""
Factorial of a number
Title - Calculate the factorial of a number
GitHub: https://www.github.com/lucastrogo
"""


def fat(num):
    """Calculate the factorial of a number"""
    ret = 1
    for elem in range(1, num + 1):
        ret *= elem

    return ret


print(fat(5))

File: C++/Algorithms/RecursionAlgorithms/Factorial.cpp
//function to calculate the factorial of a number

#include <iostream>
using namespace std;

int factorial(int n)
{
    if(n > 1)
        return n * factorial(n - 1);
    else
        return 1;
}

int main()
{
    int n;

    // input only a positive integer to calculate the factorial of
    cout << "Enter a positive integer: ";
    cin >> n;
    cout << "Factorial of " << n << " = " << factorial(n);

    return 0;
}
